internal/gateway/schema_load.go

```diff
@@ -43,6 +44,11 @@ func loadEndpointSchema(config Config, upstream *upstreamServer) (*schema.Schema
 			if upstreamSchemaFileExists {
 				config.Log.Printf("download failed (will load cached schema version): %v", err)
 			} else {
+
+				if _, ok := err.(*json.SyntaxError); ok {
```

Comment:
So generally backend would return no result if domain is down or some status code like 404.
Parsing JSON in this case should be redundant because we know that result is not JSON.
I think we trying to solve this by XY problem. 

Problem is simple - sometimes schema cannot be downloaded and we still give some
very unfriendly message. So instead of fixing problem we try to map JSON parsing error into 
something else, but JSON parsing error is right for this - we cannot parse JSON if API is gone or if API has wrong JSON. But we can know that API is gone before we parse json

Reply:
Yeah, you are right. API is not there and I was trying to validate JSON for that 😶